generate/generate.go

```diff
@@ -39,41 +46,45 @@ func New() Generator {
 		},
 		Process: rspec.Process{
 			Terminal: false,
-			User:     rspec.User{},
 			Args: []string{
 				"sh",
 			},
-			Env: []string{
-				"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
-				"TERM=xterm",
-			},
-			Cwd: "/",
-			Capabilities: []string{
-				"CAP_CHOWN",
-				"CAP_DAC_OVERRIDE",
-				"CAP_FSETID",
-				"CAP_FOWNER",
-				"CAP_MKNOD",
-				"CAP_NET_RAW",
-				"CAP_SETGID",
-				"CAP_SETUID",
-				"CAP_SETFCAP",
-				"CAP_SETPCAP",
-				"CAP_NET_BIND_SERVICE",
-				"CAP_SYS_CHROOT",
-				"CAP_KILL",
-				"CAP_AUDIT_WRITE",
-			},
-			Rlimits: []rspec.Rlimit{
-				{
-					Type: "RLIMIT_NOFILE",
-					Hard: uint64(1024),
-					Soft: uint64(1024),
-				},
-			},
 		},
 		Hostname: "mrsdalloway",
-		Mounts: []rspec.Mount{
+	}
+
+	if *os == "linux" {
+		spec.Process.User = rspec.User{}
```

Comment:
On Wed, Aug 17, 2016 at 08:38:58AM -0700, Ma Shimiao wrote:
> >  		},
> >  		Hostname: "mrsdalloway",
> > -		Mounts: []rspec.Mount{
> > +	}
> > +
> > +	if *os == "linux" {
> > +		spec.Process.User = rspec.User{}
> 
> User is not only used for Linux, it's also used by solaris.

We could handle that in two ways.  If there is significant overlap
between some Linux and Solaris defaults, we'd want:

  if *os == "linux" || *os == "solaris" {
    spec.Process.User = rspec.User{}
    …other common setup…
  }

  if *os == "linux" {
    spec.Process.Capabilities = …
    …other Linux-specific setup…
  } else if *os == "solaris" {
    spec.Solaris.Milestone = "svc:/milestone/container:default"
    …other Solaris-specific setup…
  } else {
    return generator, fmt.Errorf("no defaults configured for %s", *os)
  }

If there is not significant overlap, it's probably easier to have:

  if *os == "linux" {
    spec.Process.User = rspec.User{}
    spec.Process.Capabilities = …
    …other Linux-specific setup…
  } else if *os == "solaris" {
    spec.Process.User = rspec.User{}
    spec.Solaris.Milestone = "svc:/milestone/container:default"
    …other Solaris-specific setup…
  } else {
    return generator, fmt.Errorf("no defaults configured for %s", *os)
  }

I think it will be easier to tell how much overlap there is once we
have a more detailed Solaris default.  I don't think setting up an
empty struct for spec.Process.User is enough to be worth claiming
support for Solaris, though, so I'd rather leave this commit as it
stands until we do have a detailed Solaris default.